interfaces/kepware/config.html

```diff
@@ -357,206 +369,335 @@
+        }
+        
+        const tagsToObjects = (tags) => {
+          const objects = [];
+          tags.forEach(tag => {
+            const {objectName, frameName, tagName} = parseId(tag.nodeId);
+            const object = objects.find(obj=>obj.name===objectName) || {name:objectName, frames:[]};
+            if (!objects.includes(object)) {
+              objects.push(object);
             }
-
-            browseResults.forEach(function(tagInfo) {
-                let newElement = realityServer.templates['tag'].content.cloneNode(true);
-                newElement.querySelector('.name').innerText = tagInfo.id;
-
-                if (typeof settings.tagsInfo[tagInfo.id] === 'undefined') {
-                    settings.tagsInfo[tagInfo.id] = {};
-                }
-
-                if (typeof settings.tagsInfo[tagInfo.id].enabled === 'undefined') {
-                    settings.tagsInfo[tagInfo.id].enabled = true;
-                }
-
-                let activeButton = newElement.querySelector('.active');
-                let editButton = newElement.querySelector('.edit');
-
-                if (settings.tagsInfo[tagInfo.id].enabled) {
-                    activeButton.classList.remove('yellow');
-                    activeButton.classList.add('green');
-                    activeButton.innerText = 'On';
-                    editButton.classList.remove('inactive');
-                    editButton.classList.add('clickAble');
-
-                    if (expandedTags[tagInfo.id]) {
-                        let tagEditsContainer = newElement.querySelector('.tagEdits');
-                        tagEditsContainer.classList.remove('displayNone');
-                    }
-
-                    editButton.addEventListener('click', function() {
-                        let tagEditsContainer = this.parentNode.querySelector('.tagEdits');
-                        if (tagEditsContainer.classList.contains('displayNone')) {
-                            tagEditsContainer.classList.remove('displayNone');
-                            expandedTags[tagInfo.id] = true;
-                        } else {
-                            tagEditsContainer.classList.add('displayNone');
-                            delete expandedTags[tagInfo.id];
-                        }
+            const frame = object.frames.find(oFrame=>oFrame.name===frameName) || {name:frameName, tags:[]};
+            if (!object.frames.includes(frame)) {
+              object.frames.push(frame);
+            }
+            const newTag = {name:tagName, enabled:false, nodeId: tag.nodeId};
+            frame.tags.push(newTag);
+          });
+          return objects;
+        }
+        
+        const machines = tagsToObjects(realityServer.hardwareInterface.allTags);
+        const savedMachines = realityServer.hardwareInterface.settings['enabledTags'].value;
+        const sortByName = (a,b)=>a.name.toLowerCase()<b.name.toLowerCase()?-1:a.name.toLowerCase()>b.name.toLowerCase()?1:0;
```

Comment:
This would be much more readable as a multi-line statement instead of a nested ternary